zproject/dev_urls.py

```diff
@@ -18,59 +18,59 @@
 
 urls = [
     # Serve useful development environment resources (docs, coverage reports, etc.)
-    url(r'^coverage/(?P<path>.*)$',
-        serve, {'document_root':
-                os.path.join(settings.DEPLOY_ROOT, 'var/coverage'),
-                'show_indexes': True}),
-    url(r'^node-coverage/(?P<path>.*)$',
-        serve, {'document_root':
-                os.path.join(settings.DEPLOY_ROOT, 'var/node-coverage/lcov-report'),
-                'show_indexes': True}),
-    url(r'^casper/(?P<path>.*)$',
-        serve, {'document_root':
-                os.path.join(settings.DEPLOY_ROOT, 'var/casper'),
-                'show_indexes': True}),
-    url(r'^docs/(?P<path>.*)$',
-        serve, {'document_root':
-                os.path.join(settings.DEPLOY_ROOT, 'docs/_build/html')}),
+    path('coverage/<str:path>',
+         serve, {'document_root':
+                 os.path.join(settings.DEPLOY_ROOT, 'var/coverage'),
+                 'show_indexes': True}),
+    path('node-coverage/<str:path>',
```

Comment:
I think you need to use `path:path` instead of `str:path` in cases like this.


> str - Matches any non-empty string, excluding the path separator, '/'. This is the default if a converter isn’t included in the expression.

> path - Matches any non-empty string, including the path separator, '/'. This allows you to match against a complete URL path rather than a segment of a URL path as with str. 


If `str` is used paths like`node-coverage/frontend_tests` would be matched `node-coverage/frontend_tests/zjsunit/`, `node-coverage/frontend_tests/zjsunit/index.js.html`etc  would not be. 

I am a bit surprised that this didnt break any tests. I guess most likely because these are  urls just for our dev environment.

Reply:
I will fix this ASAP, and I'll check if other changed urls don't have the same problem.

Thanks for checking, @hackerkid!

Reply:
Yeah, that looks to be correct.  And yeah, it didn't break tests because we don't want to require coverage data to have been generated in the development environment to run tests (and we're just serving that directory).

Reply:
There is one problem with <path:path> - it accepts NON-EMPTY string, and previously we had `.*`, so empty strings were allowed. I've checked it, and replacing it in this place doesn't break tests, but changing `re_path(r'^user_avatars/(?P<path>.*)$...' to `path(user_avatars/<path:path>...)` broke one test. Should we use this path: functionality, or just stay with re_path, so empty strings are allowed?

Reply:
Well, what's the test?  It could just be a test for our error handling for invalid input having a different error now.

I don't think that URL in particular should have a use case for an empty path, certainly.